src/main/java/com/avispl/symphony/dal/communicator/lg/lcd/LgLCDDevice.java

```diff
@@ -949,18 +1022,15 @@ private void populateMonitoringAndControllingData() throws InterruptedException
 				}
 				//Count the number of requests in one polling cycle.
 				currentCommandIndex++;
-
 				//Submit thread to fetch data
 				devicesExecutionPool.add(fetchingDataExSer.submit(() -> {
 					retrieveDataByCommandName(controllingCommand.getCommandNames(), param, controllingCommand);
 				}));
-				int finalIndexFuture = i;
-
 				// The thread responsible for checking the ExecutorService waits until the defaultConfigTimeout period has elapsed.
 				// If the Future is not completed at that point, the thread will cancel it
 				manageTimeOutWorkerThread = timeoutManagementExSer.submit(() -> {
 					int timeoutCount = 1;
-					while (!devicesExecutionPool.get(finalIndexFuture).isDone() && timeoutCount <= defaultConfigTimeout) {
+					while (!devicesExecutionPool.get(devicesExecutionPool.size() - 1).isDone() && timeoutCount <= defaultConfigTimeout) {
```

Comment:
the 1 number should be stored as the constant: int ORDINAL_TO_INDEX_CONVERT_FACTOR = 1;